eras/babbage/impl/src/Cardano/Ledger/Babbage/TxBody.hs

```diff
@@ -270,16 +270,15 @@ data Datum era
 
 instance Era era => ToCBOR (Datum era) where
   toCBOR d = encode $ case d of
-    NoDatum -> Sum NoDatum 0
-    DatumHash dh -> Sum DatumHash 1 !> To dh
-    Datum d' -> Sum Datum 2 !> To d'
+    DatumHash dh -> Sum DatumHash 0 !> To dh
+    Datum d' -> Sum Datum 1 !> To d'
+    NoDatum -> error "NoDatum should always be omitted, not encoded"
```

Comment:
Why error? Would something like `OmitC` work?

It is a noop to serialize `NoDatum`, right? In this kind of place there is gotta be a way to avoid a partial function

Reply:
`OmitC` is a much better choice, thanks!